.gitignore

```diff
@@ -0,0 +1,3 @@
+target
+.idea
+
```

Comment:
This repo might benefit from a `.dockerignore` file as well.

I don't think the `build.sbt` or `codeclimate-scalastyle-assembly-0.1.0.jar` files need to be within the final image.

Reply:
`codeclimate-scalastyle-assembly-0.1.0.jar` is required - it has all the logic for the engine. It is copied as "/usr/src/app/engine-core.jar".
Only files in "src/main/resources/docker" and this jar file are included in the docker image.

Reply:
I think once it's COPY'd in, as `/usr/src/app/engine-core.jar` you can leave the original jar out of the final image.

So if a `.dockerignore` file included `codeclimate-scalastyle-assembly-0.1.0.jar`, that would remove the duplicate file, right?

Reply:
Dockerfile only copies these files:
```
COPY codeclimate-scalastyle-assembly-0.1.0.jar /usr/src/app/engine-core.jar
COPY src/main/resources/docker /usr/src/app
COPY src/main/resources/docker/engine.json /
```

Thus `codeclimate-scalastyle-assembly-0.1.0.jar` is included only once in the docker image.

Adding it to `.dockerignore` will break the build process because Docker wouldn't be able to find this jar.

Reply:
Ahh, I see. You're right. Thought there was a broader `COPY` further down, but after looking again, there isn't. LGTM.

Reply:
@dblandin - if this is good, can you please merge?